exporter/splunkhecexporter/buffer.go

```diff
@@ -61,7 +61,7 @@ func (b *bufferState) Close() error {
 
 // accept returns true if data is accepted by the buffer
 func (b *bufferState) accept(data []byte) (bool, error) {
-	if len(data)+b.rawLength > int(b.maxEventLength) {
+	if len(data) > int(b.maxEventLength) {
```

Comment:
This will cause an unnecessary batch flush every time this is encountered.  This check should not be inside the `accept`. Let's hold off on this for now. I'd like to simplify the compression logic first, which is dependent on this in some way.